tools/boost_names.py

```diff
@@ -0,0 +1,171 @@
+    Module('math', 'Math Common Factor', []),
+    Module('math', 'Math Octonion', []),
+    Module('math', 'Math Quaternion', []),
+    Module('math', 'Math/Special Functions', []),
+    Module('math', 'Math/Statistical Distributions', []),
+    Module('bind', 'Member Function', []),
+    Module('algorithm', 'Min-Max', []),
+    Module('numeric', 'Odeint', []),
+    Module('utility', 'Operators', []),
+    Module('core', 'Ref', []),
+    Module('utility', 'Result Of', []),
+    Module('algorithm', 'String Algo', []),
+    Module('core', 'Swap', []),
+    Module('', 'Tribool', []),
+    Module('numeric', 'uBLAS', []),
+    Module('utility', 'Value Initialized', []),
+]
+
+# Cannot find the following modules in the documentation of boost
+not_modules = ['beast', 'logic', 'mp11', 'winapi']
+
+def eprint(message):
+    print(message, file=sys.stderr)
+
+def get_library_names(jamfile):
+    libs = []
+    with open(jamfile) as jamfh:
+        jam = jamfh.read()
+        res = re.finditer(r'^lib[\s]+([A-Za-z0-9_]+)([^;]*);', jam, re.MULTILINE | re.DOTALL)
+        for matches in res:
+            print(matches.group(0))
+            if ':' in matches.group(2):
+                libs.append(matches.group(1))
+    return libs
+
+def exists(modules, module):
+    return len([x for x in modules if x.dirname == module.dirname]) != 0
+
+def get_modules(init=extra):
+    modules = init
+    for directory in os.listdir(LIBS):
+        if not os.path.isdir(os.path.join(LIBS, directory)):
+            continue
+        if directory in not_modules:
+            continue
+        jamfile = os.path.join(LIBS, directory, 'build', 'Jamfile.v2')
+        if os.path.isfile(jamfile):
+            libs = get_library_names(jamfile)
+        else:
+            libs = []
+        if directory in manual_map.keys():
+            modname = manual_map[directory]
+        else:
+            modname = directory.replace('_', ' ').title()
+        modules.append(Module(directory, modname, libs))
+    return modules
+
+def get_modules_2():
+    modules = []
+    for (root, dirs, files) in os.walk(LIBS):
+        for f in files:
+            if f == "libraries.json":
+                projectdir = os.path.dirname(root)
+
+                jamfile = os.path.join(projectdir, 'build', 'Jamfile.v2')
+                if os.path.isfile(jamfile):
+                    libs = get_library_names(jamfile)
+                else:
+                    libs = []
+
+                # Get metadata for module
+                jsonfile = os.path.join(root, f)
+                with open(jsonfile) as jsonfh:
+                    boost_modules = json.loads(jsonfh.read())
+                    if(isinstance(boost_modules, dict)):
+                        boost_modules = [boost_modules]
+                    for boost_module in boost_modules:
+                        modules.append(Module(boost_module['key'], boost_module['name'], libs))
+                        #print(boost_module['name'])
```

Comment:
__[flake8]__
* [E265] block comment should start with '# '

<!-- Comment by [SideCI](https://sideci.com) -->